src/NUnitEngine/nunit.engine/Runners/ProcessRunner.cs

```diff
@@ -240,7 +240,7 @@ protected override void Dispose(bool disposing)
                     log.Error(ExceptionHelper.BuildMessageAndStackTrace(ex));
                 }
 
-                if (_agent != null && _agency.IsAgentRunning(_agent.Id))
+                if (_agent != null && _agency.IsAgentRunning(_agent.Id, out var process))
```

Comment:
I can't tell what the type of process is by inspecting this line.

Reply:
Note: that's a hypothetical "I"! Obviously, __I__ do know what it is, but someone who does not might think, for example, that it's `bool process; // if true, then process this.` :smile:

Reply:
What is the benefit to knowing what the type name is while reading this block of code? Hypothetically, it might not be the process class but rather some kind of info struct with an ExitCode property, or even a `bool` variable if you don't read the code below.

In the scenarios in which I picture myself beginning to care what type it is, I'm always looking at how the value is used and where the value came from originally, which means I'm not just reading this file anymore. But I bet I'm missing something.

Reply:
When reading someone else's code, I'm always trying to understand what's going on. Being primarily a static typing kind of guy, knowing the Type makes sense to me. So, when I see this, I want to page around and find the declaration of the method so as to know the Type. That takes time.

That's why our guidelines (IIRC) say to use var only when the Type is clear, i.e. clear to the reader, since it's obviously clear to the compiler.

Reply:
Okay, happy to help save time!